### DISCUSSION TITLE
Welcome to Reflex Discussions!

### DISCUSSION THREAD

**Discussion by gustavopsantos** [gustavopsantos/Reflex#8] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by krisrok** [comment]
Great to see you added ways to communicate without creating official issues -- i often miss this 👍 

Thanks for your work on Reflex. I'm currently evaluating different DI methods for the kind of gamifications I'm working on. I've used Extenject before, and currently I'm implementing VContainer in a new codebase. But as it's in its early stages I'm pretty tempted to try out Reflex instead :) I like the idea of nested scopes in VContainer though.

From my perspective I think what this project is lacking right now is a nicer structured documentation for a quick overview as a balance between the lean Readme.md and the sample implemenation you already provide.

And finally: What was your main intent in building Reflex instead of using one of the other solutions? Are there features on the roadmap or do you want to keep it as simple as possible?

**Reply by gustavopsantos** [reply]
Hello, I'm glad you are taking a look at Reflex, and as well I agree it desires a nicer structured documentation, Im taking a look into some auto generation documentation based on code comments/attributes.

`What was your main intent in building Reflex instead of using one of the other solutions?`
I've used Extenject before, and there was a load of APIs in there, allowing things that goes counter my idea of code architecture.
Also, if you needed something, it was too hard to find it, and add it.
Thats why I implemented Reflex, readable, fast and simple.
But also I took a look into VContainer and I think it maybe nice to have some sort of scoped containers, and that is on my roadmap for sure 😉

**Comment by dadahsueh** [comment]
Regarding VContainer, I am curious about migrating and Reflex equivalents of some VContainer uses. Particularly since Unity's [Boss Room Sample](https://github.com/Unity-Technologies/com.unity.multiplayer.samples.coop) uses VContainer, I tried to use that to test replacing VContainer with Reflex. Ran in to issues with [ConnectionManager](https://github.com/Unity-Technologies/com.unity.multiplayer.samples.coop/blob/main/Assets/Scripts/ConnectionManagement/ConnectionManager.cs) @ `m_Resolver.Inject(connectionState);`, is it replaceable with AttributeInjector? Thanks!!

**Reply by gustavopsantos** [reply]
Hello! @dadahsueh, if `connectionState` is a POCO object, you should be able to replace `m_Resolver.Inject(connectionState);` with `AttributeInjector::Inject(connectionState, desiredContainer)`
If `connectionState` is a `gameObject`, then you should use the most suitable manual injection method from `GameObjectInjector`.

**Comment by dadahsueh** [comment]
Suggestions for `ReflexMenuItems`
- Debugger menu item path from `Reflex/Debugger` to `Window/Reflex Debugger %e`
  - Reasoning: since it is a debugger and it has a keyboard shortcut, maybe place it with other debuggers like UniTask Tracker, etc.
- Settings/ProjectScope path from `Assets/Create/Reflex/xxx` `priority = 0` to `Assets/Create Reflex/xxx` `priority = 3`
  - Reasoning: creating folders and C# scripts are muscle memory, so use a separate create menu item with priority 3 because projects might want to have creation menu items before it?
- SceneScope from `priority = 0` to something not 0 or none
  - Reasoning: creating empty gameobjects are muscle memory, and SceneScope is basically just empty gameobject with a component.

**Comment by gustavopsantos** [comment]
@dadahsueh Great suggestions, however I think it can be even improved.
1. Instead of `Window/Reflex Debugger` sounds to be more suitable to put it togheter with other unity debuggers, under `Window/Analysis/Reflex Debugger %e`
2. Good point, I agree with you, Im gonna remove priority setting, so it renders at the end of the list.
3. I also agree here, Im gonna remove priority setting, so it renders at the end of the list.

I'll try to push these changes end of the week ;)
